simulation/test_swarm.cpp

```diff
@@ -558,17 +558,14 @@ TORRENT_TEST(block_uploaded_alert)
 			if (auto at = lt::alert_cast<lt::add_torrent_alert>(a))
 			{
 				// init blocks vector, MUST happen before any block_uploaded_alert alerts
-				std::shared_ptr<lt::torrent> torrent = at->handle.native_handle();
-				std::shared_ptr<const torrent_info> torrent_info = torrent->get_torrent_copy();
+				auto torrent = at->handle.native_handle();
+				auto torrent_info = &torrent->torrent_file();
```

Comment:
hm. do you still need the ``torrent`` object because of ``block_size()``?
you could actually hard code the block size to ``0x4000`` as well. Originally I intended to make the block size configurable, but the value of that ended up not warrant the complexity, so by now that block size of 0x4000 is quite engrained in libtorrent (and hard coding it here would not contribute significantly to the difficulty of changing this in the future)